lib/src/main/java/com/knuddels/jtokkit/EncodingFactory.java

```diff
@@ -144,7 +144,11 @@ static Pattern compileRegex(String patternString, boolean caseInsensitive) {
             return Pattern.compile(patternString, flags);
         } catch (IllegalArgumentException exception) {
             // Workaround for Android where an IllegalArgumentException is thrown when using UNICODE_CHARACTER_CLASS
-            return Pattern.compile(patternString);
+            int flags = 0;
```

Comment:
`Pattern.compile(patternString)` delegates to `Pattern.compile(patternString, 0)`